mantidproject/mantid
# Add check for zero vector in angle calculation of DetectorInfo

**Description of work.**

This Commit adds a check in detector info that catching runtime errors from V3D when calculation angles. This error appears often when working with polaris data which features detectors with l1 and l2 equal to zero but are not monitors.

**Report to:** Helen Playford

**To test:**

The script below loads a raw polaris workspace, the instrument has dummy detectors without geometry but not marked as monitors. The `ConvertUnits` algorithm should give warnings that angles can not be calculated, but it should still run. 
```
from mantid.simpleapi import *

Load(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxpolaris\instrument\data\cycle_17_3\POLARIS00100017.nxs', OutputWorkspace='POLARIS00100017')
ConvertUnits(InputWorkspace='silicon_raw', OutputWorkspace='merged_ws', Target='MomentumTransfer')
```

Fixes #29103 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
